src/MusicStore/Program.cs

```diff
@@ -37,62 +39,96 @@ public static void Main(string[] args)
+
+                    for (int i = 0; i < threshholds.Length; i++)
                     {
-                        maxRequestTime = requestTimeElapsed;
+                        int iterationRequests = threshholds[i] - totalRequests;
+                        eventSource.RequestBatchBegin(i, iterationRequests);
+                        MeasureThroughput(client, iterationRequests, out double msRequired, out double minRequestTime, out double maxRequestTime, out double requestPerSec);
+                        eventSource.RequestBatchEnd(i, iterationRequests, (int)msRequired, minRequestTime, maxRequestTime);
+                        totalTimeMs += msRequired;
+                        Console.WriteLine("{0,5:D}-{1,5:D}   {2,18:D}   {3,5:F}   {4,15:F}   {5,15:F}",
+                                           totalRequests + 1, totalRequests + iterationRequests, (int)totalTimeMs, requestPerSec, minRequestTime, maxRequestTime);
+                        totalRequests += iterationRequests;
                     }
 
-                    // Rolling average of request times
-                    averageRequestTime = (averageRequestTime * ((i - 1.0) / i)) + (requestTimeElapsed * (1.0 / i));
+                    Console.WriteLine();
+                    Console.WriteLine("Tip: If you only care about startup performance, use the -skipSteadyState argument to skip these measurements");
                 }
 
-                Console.WriteLine("Steadystate min response time: {0}ms", minRequestTime);
-                Console.WriteLine("Steadystate max response time: {0}ms", maxRequestTime);
-                Console.WriteLine("Steadystate average response time: {0}ms", (int)averageRequestTime);
             }
 
-            Console.WriteLine();
-
+            
             VerifyLibraryLocation();
         }
 
+       
+        
+
+        private static void MeasureThroughput(HttpClient client, int countRequests, out double msRequired, out double minRequestTime, out double maxRequestTime, out double requestPerSec)
+        {
+            var requestGroupTimer = Stopwatch.StartNew();
+            var requestTime = Stopwatch.StartNew();
+            minRequestTime = long.MaxValue;
+            maxRequestTime = long.MinValue;
+
+            for (int i = 1; i <= countRequests; i++)
+            {
+                requestTime.Restart();
+                var response = client.GetAsync("http://localhost:5000").Result;
+                requestTime.Stop();
+
+                var requestTimeElapsed = requestTime.ElapsedTicks*1000.0/Stopwatch.Frequency;
+                if (requestTimeElapsed < minRequestTime)
+                {
+                    minRequestTime = requestTimeElapsed;
+                }
+
+                if (requestTimeElapsed > maxRequestTime)
+                {
+                    maxRequestTime = requestTimeElapsed;
+                }
+            }
+
+             msRequired = requestGroupTimer.ElapsedTicks*1000.0/Stopwatch.Frequency;
```

Comment:
msRequired - Not getting what is required, looks like it's the time taken for the group, could it use a rename?

Reply:
that's correct, I'll rename